main.py

```diff
@@ -1,132 +1,379 @@
 
 
-import argparse
-from sklearn_extra.cluster import KMedoids
+import os.path
+# import time
+
+import numpy
 import pandas as pd
 import sys
-DATABUFFER = 268435456 # size of quarter of a gigabyte
+import psutil
+import math
+from sklearn_extra.cluster import KMedoids
+# from sklearn.compose import ColumnTransformer
+# from sklearn.preprocessing import OneHotEncoder
+from sklearn.preprocessing import LabelEncoder
 
-# def decide_row_num_halt(path,cols=None):
-#     if path[-5:] == ".xlsx" or path[-4:] == ".xls":
-#         temp = pd.read_excel(path,usecols=cols,nrows=1)
-#     elif path[-4:] == ".csv":
-#         temp=pd.read_csv(path,usecols=cols,nrows=1)
-#     else:
-#         return
-#     row_num= DATABUFFER/ sys.getsizeof(temp)
 
+def load_data_xls(path, startRows = None,
+                  endRows = None, cols = None, 
+                  sheet = 0):
+    header = pd.read_excel(path, usecols = cols, nrows = 1, sheet_name = sheet)
+    temp = pd.read_excel(path,
+                         usecols = cols,
+                         nrows = endRows - startRows if
+                         startRows else endRows, skiprows = startRows,
+                         sheet_name = sheet)
+    temp.columns = header.columns
+    return temp
+
+
+def load_data_csv(path, startRows = None, endRows = None, cols = None):
+    header = pd.read_csv(path, usecols = cols, nrows = 1)
+    temp = pd.read_csv(path,
+                       usecols = cols,
+                       nrows = endRows - startRows if
+                       startRows else endRows,
+                       skiprows = startRows,
+                       thousands = ",")
+    temp.columns = header.columns
+    return temp
 
-def load_data(path,startrows=None,endrows=None,cols=None):
-    # load_data(path, startRow, endRow, cols) -> data
-    # path – path to file startRow – first row to load
+
+def load_data_type(path,
+                   startRows = None,
+                   endRows = None,
+                   cols = None,
+                   sheet = 0):
+    if path[-5:] == ".xlsx" or path[-4:] == ".xls":
+        return load_data_xls(path, startRows, endRows, cols, sheet)
+    elif path[-4:] == ".csv":
+        return load_data_csv(path, startRows, endRows, cols)
+
+
+def load_data(path, startRows = None, endRows = None, cols = None, sheet = 0):
+    # load_data(path, startRows, endRow, cols) -> data
+    # path – path to file startRows – first row to load
     # endRow – last row to load, or the last of file, the minimum
-    # cols – columns to load, there may be columns we would not want to load as they are useless
+    # cols – columns to load, there may be columns we would not want to load
+    # as they are useless
     # data – return the loaded partial file
+    try:
+        return load_data_type(path, startRows, endRows, cols, sheet)
+    except:
+        return pd.DataFrame()
```

Comment:
Not sure what this function does

Reply:
this function is to catch the load when it gets a startRow higher than the amount of rows it has, it then returns an empty dataFrame.
i could do it nicer
what im trying to avoid is loading too many lines (for example with a csv with a 10 million lines and overloading the memory)